cogs/tags.py

```diff
@@ -542,6 +547,64 @@ def top_three_tags(self, db):
 
+    @tag.command(pass_context=True, no_pm=True)
+    @checks.sensei_or_mod_or_permissions(manage_messages=True)
+    async def transfer(self, ctx, tag_name, user: discord.Member):
+        """
+        Transfers a tag that you have created to another user.
+        This can be done by the creator of the tag. Cannot transfer 
+        if the user being transfered to is over the tag limit.
+        """
+        lookup = tag_name.lower().strip()
+        server = ctx.message.server
+        try:
+            tag = self.get_tag(server, lookup, redirect=False)
+        except RuntimeError as e:
+            return await self.bot.say(e)
+        
+        adminRoleName = self.bot.settings.get_server_admin(server)
+        modRoleName = self.bot.settings.get_server_mod(server)
+    
+        adminRole = discord.utils.get(ctx.message.server.roles, name=adminRoleName)
+        modRole = discord.utils.get(ctx.message.server.roles, name=modRoleName)
+    
+        sensei = discord.utils.get(ctx.message.server.roles, name="Sensei")
+        
+        #checks for mod/admin/owner of the tag
+        if tag.owner_id != ctx.message.author.id and adminRole not in ctx.message.author.roles and modRole not in ctx.message.author.roles:
+            await self.bot.say("Only the tag owner can transfer this tag.")
+            return
+
+        #checks if the user in question has permissions to create tags
+        if sensei not in user.roles and adminRole not in user.roles and modRole not in user.roles:
+            await self.bot.say("The person you are trying to transfer to cannot create commands.")
+            return
+
+        #checks if the user has exceeded the tag limit
+        if await self.user_exceeds_tag_limit(ctx.message.server, user):
+            await self.bot.say("The person you are trying to transfer a tag to already has too many commands!")
+            return
+
+        await self.bot.say("{} please confirm by saying \"yes\" that you would like to receive "
+                           "this tag from {}.".format(user.mention, ctx.message.author.mention))
+        response = await self.bot.wait_for_message(timeout=15, author=user)
+        if not response:
+            await self.bot.say("No comfirmation from {}. Transfer has been cancelled.".format(user.name))
+            return
+        elif response.channel == ctx.message.channel:
+            if response.content.startswith("yes"):
+                #The user has answered yes; transfering tag
+                db = self.config.get(tag.location)
+                tag.owner_id = user.id
+                await self.config.put(tag.location, db)
+                await self.bot.say("Tag successfully transferred from the current owner "
+                                   "to {}.".format(user.mention))
+            else:
+                await self.bot.say("Tag has been rejected by {}. Transfer has been "
+                                   "cancelled.".format(user.name))
+        else:
```

Comment:
What you had before was better. This will be messy if the user is in mid-sentence, typing in another channel.  See next comment.